src/constraints.jl

```diff
@@ -150,25 +150,28 @@ function JuMP.delete(model::JuMP.AbstractModel, cref::LogicalConstraintRef)
     @assert JuMP.is_valid(model, cref) "Logical constraint does not belong to model."
     delete!(_logical_constraints(model), JuMP.index(cref))
     _set_ready_to_optimize(model, false)
-    return 
+    return
 end
 
 ################################################################################
 #                              Disjunct Constraints
 ################################################################################
 function _check_expression(expr::Ex) where {Ex <: JuMP.AbstractJuMPScalar}
     vars = Set{JuMP.variable_ref_type(expr)}()
-    _interrogate_variables(v -> push!(vars, v), expr) 
-    if any(JuMP.is_binary.(vars)) || any(JuMP.is_integer.(vars))
-        error("Disjunct constraints cannot contain binary or integer variables.")
+    _interrogate_variables(v -> push!(vars, v), expr)
+    if any(JuMP.is_binary.(vars))
+        error(
+            "Disjunct constraints should not contain binary variables. " *
+            "Instead, use Logical variables to express any binary decisions."
+        )
```

Comment:
I am not sure this is sufficient since logical variables cannot be used inside algebraic constraints with regular JuMP variables. I think allowing both binary and integer variables is the way to go. Naturally, this has implications for the reformulations where we currently assume any binary variable is a disjunct indicator variable. However, we can instead check if the binary variable corresponds to an indicator variable via a mapping in `GDPData`.

Reply:
My point here is that if the user wants to have a binary variable inside a disjunct constraint, this should be coming from a nested disjunction. The user should model their problem as a proper GDP. Perhaps a better error message would be:

"Disjunct constraints can only contain continuous or integer variables, but not binary variables. If you are modeling binary decisions, you should define these in a nested disjunction."

I am still not convinced that we should allow binary variables in a disjunction constraint. I see integer variables being allowed, as these can have a physical meaning (e.g., number of trucks purchased), but not binary variables.

Reply:
@pulsipher I have updated the error message text to:
```julia
        error(
            "Disjunct constraints can only contain continuous or integer variables, " *
            "not binary variables. If you are modeling binary decisions, " *
            "you should express these using the Logical variables that serve as indicators " *
            "to each of the disjuncts. If these are lower-level decisions to the disjunct " *
            "being defined, these can be modeled using nested disjunctions."
        )
```

What are your thoughts on this issue of allowing binaries?

Reply:
I still suspect that are situations where it is difficult to reformulate into nested disjunctions. For instance, certain conic sets that rely on binary variables or perhaps a piecewise linear function. These could be reformulated into nested disjunctions, but that puts a lot of burden on the user. 

However, for now adding integer variables is still a win and the error message makes things more clear. We can wait for someone to come up with a compelling use case for allowing binary variables.